File: board_dsl/src/main/java/com/dooho/board/repository/UserRepository.java
package com.dooho.board.repository;

import com.dooho.board.entity.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Transactional;

@Repository
@Transactional
public interface UserRepository extends JpaRepository<UserEntity, String> {
    boolean existsByUserEmailAndUserPassword(String userEmail, String userPassword);

    UserEntity findByUserEmail(String userEmail);

    boolean existsByUserNickname(String userNickname);

}

File: board_spring/src/main/java/com/dooho/board/repository/UserRepository.kt
package com.dooho.board.repository

import com.dooho.board.entity.UserEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional

@Repository
@Transactional
interface UserRepository : JpaRepository<UserEntity?, String?> {
    fun existsByUserEmailAndUserPassword(userEmail: String?, userPassword: String?): Boolean
    fun findByUserEmail(userEmail: String?): UserEntity?
    fun existsByUserNickname(userNickname: String?): Boolean
}